File: 2005-check-if-all-the-integers-in-a-range-are-covered/2005-check-if-all-the-integers-in-a-range-are-covered.py
class Solution:
    def isCovered(self, ranges: List[List[int]], left: int, right: int) -> bool:
        covered = True
        for i in range(left, right + 1):
            c = False
            for l, r in ranges:
                if l <= i <= r:
                    c = True
                    break
            covered = covered and c
        return covered

File: 2005-check-if-all-the-integers-in-a-range-are-covered/2005-check-if-all-the-integers-in-a-range-are-covered.java
class Solution {
    public boolean isCovered(int[][] ranges, int left, int right) {
        boolean covered = true;
        for (int i = left; i <= right; i++) {
            boolean c = false;
            for (int[] range : ranges) {
                if (range[0] <= i && i <= range[1]) {
                    c = true;
                    break;
                }
            }
            covered = c && covered;
        }
        return covered;
    }
}